lib/Server.js

```diff
@@ -406,7 +407,7 @@ class Server {
         ? parseInt(process.env.WEBPACK_DEV_SERVER_PORT_RETRY, 10)
         : 3;
 
-    return pRetry(() => getPort(basePort), {
```

Comment:
I am not sure `retry` is going to help a lot as the `getPort()` (and portfinder) before is going to try every port up to the maximum port number anyway, but it was introduced on purpose in [this commit](https://github.com/webpack/webpack-dev-server/pull/1993) back in 2019, so there might be a reason.